### DISCUSSION TITLE
Edge caching and cookies

### DISCUSSION THREAD

**Discussion by liegeandlief** [vercel/community#1440] [Help]
Assuming that my response has a `cache-control` header which would cause it to be cached at the edge:

Is the `cookie` request header value used as part of the cache key?

Also if the request has no `cookie` header and the response contains a `set-cookie` header will the response still be cached? Is it the responsibility of my application to ensure the response isn't cached in this situation.

In general I think the docs around the behaviour of the edge network cache could be improved. Some things can be worked out by trial-and-error but without comprehensive documentation there is always a possibility that incorrect assumptions about cache behaviours are made.

**Comment by fightingmonk** [comment]
I have a similar same question and don't see a clear answer in the docs. 

My use case: site's landing page is server-side rendered and contains customizations if the request comes from an authenticated user. 

1. For requests that don't have an auth cookie, I want Edge to server-side cache the static page and return the cached content to all cookie-less requests. 
2. For requests to DO have an auth cookie, I want the request forwarded to the page's SSR function and optionally set a client-side cache header.
